src/fingerprint.rs

```diff
@@ -292,26 +292,67 @@ fn lookup_all_fingerprint_dirs(dir: &Path) -> impl Iterator<Item = DirEntry> {
         })
 }
 
+fn is_custom_toolchain(toolchain: &str) -> bool {
+    if toolchain.is_empty() {
+        // unsure
+        return false;
+    }
+
+    let is_named_channel = ["stable", "beta", "nightly"].iter().any(|channel| toolchain.starts_with(channel));
+    if is_named_channel {
+        return false;
+    }
+
+    // versioned toolchain: 1.60 or 1.60.0
+    let first_segment = toolchain.split_once('-').map_or(toolchain, |(first, _)| first);
+    let mut number_segments = 0;
+    let all_numbers = first_segment.split('.').all(|s| {
+        number_segments += 1;
+        !s.is_empty() && s.chars().all(|c| c.is_ascii_digit())
+    });
+    let is_versioned_toolchain = all_numbers && (number_segments == 2 || number_segments == 3);
+    if is_versioned_toolchain {
+        return false;
+    }
+
+    true
+}
+
 fn lookup_from_names<'a>(
     iter: impl Iterator<Item = Option<impl AsRef<str>>>,
 ) -> Result<HashSet<u64>, Error> {
     iter.map(|x| {
         let args = x
+            .as_ref()
             .into_iter()
             .map(|toolchain| format!("+{}", toolchain.as_ref()))
             .chain(Some("-vV".to_string()));
         let out = Command::new("rustc").args(args).output()?;
 
         if !out.status.success() {
-            bail!(String::from_utf8_lossy(&out.stdout).to_string());
+            let toolchain = x.as_ref().map_or("", |t| t.as_ref());
+            if is_custom_toolchain(toolchain) {
+                return Ok(None);
+            }
+
+            let err = if out.stdout.is_empty() {
+                out.stderr
+            } else {
+                if !out.stderr.is_empty() {
+                    warn!("stderr from rustc: {}", String::from_utf8_lossy(&out.stderr));
+                }
+                out.stdout
+            };
```

Comment:
this is slightly complicated so we can preserve the old behavior of printing stdout if it's present - not sure why that was chosen to begin with, happy to switch to using stderr unconditionally if you think that's better.